pkg/devcontainer/setup/setup.go

```diff
@@ -144,6 +152,89 @@ func ChownWorkspace(setupInfo *config.Result, log log.Logger) error {
 	return nil
 }
 
+func EnsureCorrectUID(setupInfo *config.Result, log log.Logger) error {
+	// get host uid and gid
+	stats, err := os.Stat(setupInfo.SubstitutionContext.ContainerWorkspaceFolder)
+	if err != nil {
+		return errors.Wrap(err, "stat container workspace folder")
+	}
+	hostUID, hostGID, err := GetUserInfo(stats)
+	if err != nil {
+		return err
+	}
+
+	// get container uid and gid
+	containerUserName := config.GetRemoteUser(setupInfo)
+	containerUser, err := user.Lookup(containerUserName)
+	if err != nil {
+		return errors.Wrap(err, "get container user")
+	}
+	containerUID, containerGID := containerUser.Uid, containerUser.Gid
+
+	// if host or container user is root or host and container uid/gid match already, do nothing
+	if hostUID == "0" || containerUID == "0" || (hostUID == containerUID && hostGID == containerGID) {
```

Comment:
What happens if the host is not root, but within the container the user is root? Aren't we trying to map the root user to a different UID, GID then?

Reply:
I think here we should just check if containerUID is 0, as that's the regular way the user is mapped inside the container
Just dropping the hostUID check

Reply:
Yes Fabian, when the hostUID is non-root and the containerUID is 0 (root), we don't do anything; else this will overwrite the UID and GID within the container to hostUID/GID making it just a normal user account within the container.